lms/services/group_info.py

```diff
@@ -0,0 +1,48 @@
+"""A service that upserts :class:`lms.models.GroupInfoService` records."""
+
+from lms.models import GroupInfo
+
+__all__ = ["GroupInfoService"]
+
+
+class GroupInfoService:
+    """
+    A callable that upserts :class:`lms.models.GroupInfoService` records.
+
+    Usage:
+
+        group_info = request.find_service(name="group_info")
+        group_info.upsert(authority_provided_id, consumer_key, request.params)
+    """
+
+    # Things we don't try and update on GroupInfo objects from params
```

Comment:
```suggestion
    # GroupInfo columns that upsert() *doesn't* update, even if they're in the given params.
```

This might make more sense just inlined into `upsert()` as a local variable or even just a literal value